Userland/Libraries/LibGfx/Filters/ColorBlindnessFilter.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2021, Antonio Di Stefano <tonio9681@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Array.h>
+#include <AK/Singleton.h>
+#include <LibGfx/Filters/ColorFilter.h>
+
+namespace Gfx {
+
+class ColorBlindnessFilter : public ColorFilter {
+public:
+    virtual ~ColorBlindnessFilter() { }
+
+    virtual const char* class_name() const override { return "ColorBlindnessFilter"; }
+#define CREATE_FILTER(name, ...)                                       \
+    static ColorBlindnessFilter* name()                                \
+    {                                                                  \
+        static auto s_name = new ColorBlindnessFilter { __VA_ARGS__ }; \
+        return s_name;                                                 \
+    }
```

Comment:
This seems a bit awkward:

- It's effectively a singleton pattern, but doesn't follow our usual implementation of singletons (static `the()`)
- I don't think singleton is a good fit here. Sure, you might only ever create one of these, but it's not a requirement and should not be enforced here on the library level.

I suggest we use a smart pointer instead, similar to how filters are used in PixelPaint (no need for the macro either):

```cpp
#include <AK/NonnullOwnPtr.h>

static NonnullOwnPtr<ColorBlindnessFilter> create_protanopia()
{
    return make<ColorBlindnessFilter>(
        .56, .44, 0,
        .55, .45, 0,
        0, .24, .76);
}
```

`PreviewWidget`'s `m_color_filter` would then be an `OwnPtr<Gfx::ColorBlindnessFilter>`.